### ISSUE TITLE
navigate() behaves differently form <a> routing

### ISSUE THREAD

**Issue by samuelstroschein** [brillout/vite-plugin-ssr#502]
### Question / Discussion

**Problem**  

`navigate()` behaves differently from `<a>` routing. Specifically, the navigate function ignores pages that are supposed to be only server side rendered via the`.page.server.tsx` name convention while `<a>` routing obeys to the convention.

**Expectation**
I expected that `navigate` behaves identical to `<a>` routing in a client-side environment. 

**Reproduction**
https://stackblitz.com/edit/vitejs-vite-ucz5ho?file=pages/index.page.tsx

**Discussion/Proposal** 

I am aware that `navigate` is supposed for client-side navigation only. Therefore, I am unsure whether this is a bug or on purpose. Regardless, the behaviour seems counter-intuitive. I expect that <a> routing performs client side routing too (correct me if I am wrong here). 

I am currently using what seems to me like a workaround: 
```tsx
export function Page() {
        // onMount because window is not available in a server-side environment....
	onMount(() => {
		// server side routing trigger
		// see https://vite-plugin-ssr.com/navigate#page-content
		window.location.href = "documentation/intro";
	});
}
```

1. I propose that navigate behaves identical to <a> routing in a client-side environment. 
2. (Open a different issue?) VPS provides routing. It seems like `navigate` should handle server-side routing and client-side routing for the user.

**Comment by brillout** [comment]
`navigate()` and `<a>` clicks actually use the same callback, so they do behave the same.

What are the reproduction steps leading to a actual/expected discrepancy?

**Comment by samuelstroschein** [comment]
I created a video. Let me know if the video format works for you/is beneficial as a maintainer. 

https://user-images.githubusercontent.com/35429197/200197087-cf7c7790-174a-4830-a0d9-f2e0eb168672.mp4

**Comment by brillout** [comment]
Fix released in `0.4.45`. Thanks for the video 🙂.
